### DISCUSSION TITLE
Preference for complicated commits

### DISCUSSION THREAD

**Discussion by Tyler-Keith-Thompson** [wwt/SwiftCurrent#75] [Q&A] [answered]
I think I've found a way to make SwiftUI work without needing `AnyView` and it looks kinda promising. I'm not ready to create an issue for it yet because I'm still playing with the details. I'd like to be able to contribute that in a way that doesn't leave people trying to understand and end result for days on end. How does the team think I should go about this?

Some Ideas:
- Draft PR where I comment frequently and explain what I'm doing and why
- Discussion where I pontificate on the direction i'm headed with the PR
- Big commit 2 days from now that the team has to catch up with
- Something else

**Comment by morganzellers** [comment]
I think a draft PR with comments would be easiest to digest personally so that gets my vote.

**Comment by Richard-Gist** [comment]
I think I would like all of the above. 

- A Draft PR with frequent commits explaining what you are doing and why.
- Add to this discussion or a new discussion about the high level thing you are attempting (call out the impact to the public API).
- And then when you are ready to merge in promote your PR and add review comments on the really important parts that you think could be confusing.

With that said, I can roll with most things, so do what makes sense to you.  If you start a draft PR, make sure to call us out on anything you are wanting specific feedback about.

**Comment by brianlombardo** [comment, marked as answer]
In order to best utilize your time and get work merged in faster, we recommend:

+ a Discussion to ensure that we are aligned on direction

+ a PR that contains frequent commits with comments
